src/OrbitGl/LineGraphTrack.cpp

```diff
@@ -98,21 +112,28 @@ void LineGraphTrack<Dimension>::DrawSeries(PrimitiveAssembler& primitive_assembl
 
     if (aggr.GetAccumulatedEntry() == nullptr) {
       aggr.SetEntry(prev_time, curr_time, curr_normalized_values);
+      last_entry_values = curr_normalized_values;
       continue;
     }
-    // If the accumulated entry still fits in the pixel
-    if (aggr.GetAccumulatedEntry()->end_tick < next_pixel_start_ns) {
+
+    // If the current data point fits into the same pixel as the entry we are currently
+    // accumulating.
+    if (curr_time < next_pixel_start_ns) {
       // Add the current data to accumulated_entry
       aggr.MergeDataIntoEntry(prev_time, curr_time, curr_normalized_values);
+      last_entry_values = curr_normalized_values;
```

Comment:
Nit: Can you move this line outside of the if-else statement (line 137-138)? I initially didn't understand why you were doing this after MergeDataIntoEntry and I think that if you move it later it can be easier to understand.